### PR TITLE
docs: update changelog for the v2.24.0 release

### PR DESCRIPTION
**Pull request by devbww**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14110)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -44,6 +44,33 @@ have stopped (or shortly will stop) testing with MSVC 2019 too.
...
+- fix(spanner): avoid absl types in public Interval API ([#14095](https://github.com/googleapis/google-cloud-cpp/pull/14095))
+- feat(spanner): add a representation for the Spanner INTERVAL ([#14059](https://github.com/googleapis/google-cloud-cpp/pull/14059))
+- fix(spanner): delete idle sessions upon SessionPool destruction ([#14012](https://github.com/googleapis/google-cloud-cpp/pull/14012))
+
+### [Storage](/google/cloud/storage/README.md)
+
+- fix(storage): missing include ([#14052](https://github.com/googleapis/google-cloud-cpp/pull/14052))
+- fix(GCS+gRPC): MSVC build problems ([#14031](https://github.com/googleapis/google-cloud-cpp/pull/14031))
+- feat(GCS+gRPC): checksum options ([#14002](https://github.com/googleapis/google-cloud-cpp/pull/14002))
+- docs(GCS+gRPC): refactor AsyncClient comments ([#14001](https://github.com/googleapis/google-cloud-cpp/pull/14001))
```
Comment (coryan) [inline CHANGELOG.md:62]: Please remove the GCS+gRPC entries.

File: CHANGELOG.md [hunk 2]
```diff
@@ -44,6 +44,33 @@ have stopped (or shortly will stop) testing with MSVC 2019 too.
...
+
+### [Common Libraries](/google/cloud/README.md)
+
+- feat(datastore): emulator environment variable ([#14100](https://github.com/googleapis/google-cloud-cpp/pull/14100))
```
Comment (coryan) [inline CHANGELOG.md:66]: Consider a separate entry for the Datastore library.

File: CHANGELOG.md [hunk 3]
```diff
@@ -44,6 +44,33 @@ have stopped (or shortly will stop) testing with MSVC 2019 too.
...
+- docs(storagecontrol): use `GetStorageLayout()` for quickstart ([#14044](https://github.com/googleapis/google-cloud-cpp/pull/14044))
+- docs(storagecontrol): update service title ([#14042](https://github.com/googleapis/google-cloud-cpp/pull/14042))
+
+### [Google APIs interface definitions](https://github.com/googleapis/googleapis)
+
+- This release is based on definitions as of [2024-04-30T21:03:48Z](https://github.com/googleapis/googleapis/tree/5a0ae652df1ef39b7849bf2bd6ba417376ce3204)
```
Comment (dbolduc) [inline CHANGELOG.md:72]: neat

File: CHANGELOG.md (hunk 1 above)
Comment (devbww) [reply CHANGELOG.md:62]: Done.

File: CHANGELOG.md (hunk 2 above)
Comment (devbww) [reply CHANGELOG.md:66]: Done.  And did the same for "Storage Control", so there is no longer a "Common Libraries" section.

**Review by dbolduc — APPROVED** [review]

**Pull request merged**